sycl/test-e2e/KernelCompiler/kernel_compiler_sycl.cpp

```diff
@@ -11,22 +11,29 @@
 
 // -- Test the kernel_compiler with SYCL source.
 // RUN: %{build} -o %t.out
-// RUN: %{run} %t.out
-// RUN: %{l0_leak_check} %{run} %t.out
+
+// If clang++ is not on the PATH, or if sycl was compiled with GCC < 8, then
+// the kernel_compiler is not available for SYCL language.
+// Note: this 'invoking clang++' version for SYCL language support is temporary,
+// and will be replaced by the SYCL_JIT version soon.
+// DEFINE: %{available} = %t.out available
+
+// RUN: %if available %{  %{run} %t.out  %}
```

Comment:
I think this condition doesn't work as expected. Regardless of whether the executable returns 0 or 1, I get
```
# RUN: at line 21 has no command after substitutions
```
Can such a substitution be used for the `%if` construct? The [testing guide](https://llvm.org/docs/TestingGuide.html#substitutions) only mention "features" as conditions.

Reply:
I think you are right - these tests aren't running at all.  This is a problem, the post-commit testing has a pass where the invoking kernel_compiler simply isn't available. Might have to figure out a different way of excluding that pass.